app/controllers/concerns/state_file/repeated_question_concern.rb

```diff
@@ -22,13 +22,16 @@ def prev_path
     end
 
     def next_path
-      return super if params[:return_to_review].present?
+      return super if params[:return_to_review].present? && !review_all_items_before_returning_to_review
```

Comment:
this is the default action that happens if `return_to_review` is present except for this `review_all_items_before` case